### PR TITLE
chore:core: add tests for recipient in hooks

### PR DESCRIPTION
**Pull request by nmimran99**
chore:core: add tests for recipient in hooks
This PR adds tests to hooks that can have the `funder` and `recipient` as 2 different users. 
The tests are focused on the parts where the difference comes in play.

### CODE REVIEW

File: apps/smart-contracts/core/test/unit/DepositHook.test.ts [hunk 1]
```diff
@@ -156,6 +156,15 @@ describe('=> DepositHook', () => {
       expect(depositRecord.recordDeposit).calledWith(user.address, TEST_AMOUNT_AFTER_FEE)
     })
 
+    it('calls recordDeposit() if fee > 0 and funder != recipient', async () => {
+      expect(user.address).not.eq(recipient.address)
+      await depositHook
+        .connect(collateral.wallet)
+        .hook(user.address, user.address, TEST_AMOUNT_BEFORE_FEE, TEST_AMOUNT_AFTER_FEE)
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/core/test/unit/DepositHook.test.ts:163]: I think this is a mistake, shouldn't the second parameter here be recipient.address? As well as the expect statement?

**Review by ramenforbreakfast — CHANGES_REQUESTED** [review]
So we also should check that the contracts calling hooks pass the correct parameters, I see in the  `Collateral.test.ts` we've already done this (I guess you added it in the initial PR #882?), but if you look in the `PrePOMarket.test.ts`, we are still using the same account when testing hook parameters.

**Review by ramenforbreakfast — APPROVED** [review]
LGTM

File: apps/smart-contracts/core/test/unit/DepositHook.test.ts [hunk 2]
```diff
@@ -156,6 +156,15 @@ describe('=> DepositHook', () => {
       expect(depositRecord.recordDeposit).calledWith(user.address, TEST_AMOUNT_AFTER_FEE)
     })
 
+    it('calls recordDeposit() if fee > 0 and funder != recipient', async () => {
+      expect(user.address).not.eq(recipient.address)
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/unit/DepositHook.test.ts:160]:
```suggestion
      expect(user.address).not.eq(recipient.address)
      
```

File: apps/smart-contracts/core/test/unit/RedeemHook.test.ts [hunk 3]
```diff
@@ -161,6 +158,13 @@ describe('=> RedeemHook', () => {
         expect(tokenSender.send).calledWith(user.address, 1)
       })
 
+      it('calls tokenSender.send() if fee > 0 and funder != recipient', async () => {
+        expect(user.address).not.eq(recipient.address)
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/unit/RedeemHook.test.ts:162]:
```suggestion
        expect(user.address).not.eq(recipient.address)
        
```

File: apps/smart-contracts/core/test/unit/RedeemHook.test.ts [hunk 4]
```diff
@@ -161,6 +158,13 @@ describe('=> RedeemHook', () => {
         expect(tokenSender.send).calledWith(user.address, 1)
       })
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/unit/RedeemHook.test.ts:160]: add to the above test `and redeemer = recipient`

File: apps/smart-contracts/core/test/unit/RedeemHook.test.ts [hunk 5]
```diff
@@ -161,6 +158,13 @@ describe('=> RedeemHook', () => {
...
 
+      it('calls tokenSender.send() if fee > 0 and funder != recipient', async () => {
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/unit/RedeemHook.test.ts:161]:
```suggestion
      it('calls tokenSender.send() if fee > 0 and redeemer != recipient', async () => {
```

File: apps/smart-contracts/core/test/unit/WithdrawHook.test.ts [hunk 6]
```diff
@@ -153,6 +154,18 @@ describe('=> WithdrawHook', () => {
         expect(fakeTokenSender.send).calledWith(user.address, fee)
       })
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/unit/WithdrawHook.test.ts:156]: add to the above test `and funder = recipient`

**Review by XavierEkkel — APPROVED** [review]
LGTM after nits

**Pull request merged**